include/functions/conversion.hpp

```diff
@@ -15,8 +15,16 @@
 */
 
 std::string BigInt::to_string() const {
-    // prefix with sign if negative
-    return this->sign == '-' ? "-" + this->value : this->value;
+
+    std::string num_string;
+
+    for (uint64_t ull : this->magnitude)
+        while (ull > 0) {
```

Comment:
This wouldn't work. Base conversion will need to be done. I'm working on it, so you can leave this as it previously was.